### PR TITLE
Fix o11y timeouts

### PR DESCRIPTION
**Pull request by 07souravkunda**

### CODE REVIEW

File: bin/testObservability/helper/helper.js [hunk 1]
```diff
@@ -39,6 +41,7 @@ exports.pending_test_uploads = {
 };
 
 exports.debug = (text, shouldReport = false, throwable = null) => {
+  consoleHolder.log(`[ OBSERVABILITY ] ${text}`);
```
Comment (amaanbs) [inline bin/testObservability/helper/helper.js:44]: Do we need this?

File: bin/testObservability/helper/helper.js [hunk 2]
```diff
@@ -391,6 +397,7 @@ exports.launchTestSession = async (user_config, bsConfigPath) => {
       exports.debug('Build creation successfull!');
       process.env.BS_TESTOPS_BUILD_COMPLETED = true;
       setEnvironmentVariablesForRemoteReporter(response.data.jwt, response.data.build_hashed_id, response.data.allow_screenshots, data.observability_version.sdkVersion);
+      consoleHolder.log(response.data.build_hashed_id);
```
Comment (amaanbs) [inline bin/testObservability/helper/helper.js:400]: Do we need this?

File: bin/testObservability/helper/requestQueueHandler.js [hunk 3]
```diff
@@ -48,11 +48,20 @@ class RequestQueueHandler {
...
+    this.queue = [];
+    require('child_process').spawnSync('node', [require('path').join(__dirname, 'shutdown.js'), require('path').join(__dirname, 'queue.json')], {stdio: 'inherit'});
+  }
+
   shutdown = async () => {
     this.removeEventBatchPolling('REMOVING');
     while(this.queue.length > 0) {
       const data = this.queue.slice(0,BATCH_SIZE);
       this.queue.splice(0,BATCH_SIZE);
+      consoleHolder.log(this.queue.length + " the queue length ");
```
Comment (amaanbs) [inline bin/testObservability/helper/requestQueueHandler.js:64]: Do we need this?

File: bin/testObservability/helper/requestQueueHandler.js [hunk 4]
```diff
@@ -48,11 +48,20 @@ class RequestQueueHandler {
     }
   }
 
+  shutdownSync = () => {
+    this.removeEventBatchPolling('REMOVING');
+   
+    require('fs').writeFileSync(require('path').join(__dirname, 'queue.json'), JSON.stringify(this.queue));
```
Comment (amaanbs) [inline bin/testObservability/helper/requestQueueHandler.js:54]: Let's rename file `queue.json` to something more meaningful?

File: bin/testObservability/helper/shutdown.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const RequestQueueHandler = require('./requestQueueHandler');
+
+const shutdown = async () => {
```
Comment (amaanbs) [inline bin/testObservability/helper/shutdown.js:3]: Let's rename file & method names to indicate how this shutdown hook is used. Also a comment explaining it's usage.

File: bin/testObservability/reporter/index.js [hunk 6]
```diff
@@ -213,7 +215,7 @@ class MyReporter {
       const lastTest = this.current_test;
       this.current_test = test;
       test.retryOf = null;
-      test.testAnalyticsId = uuidv4();
+      test.testAnalyticsId = uuidv4()
```
Comment (amaanbs) [inline bin/testObservability/reporter/index.js:218]: Needed?

File: bin/testObservability/reporter/index.js [hunk 7]
```diff
@@ -15,6 +15,8 @@ const { v4: uuidv4 } = require('uuid');
 const { IPC_EVENTS } = require('../helper/constants');
 const { startIPCServer } = require('../plugin/ipcServer');
 
+const ipc = require('node-ipc');
```
Comment (amaanbs) [inline bin/testObservability/reporter/index.js:18]: import needed?

File: bin/testObservability/reporter/index.js [hunk 8]
```diff
@@ -199,6 +200,7 @@ class MyReporter {
         server.on(IPC_EVENTS.COMMAND, this.cypressCommandListener.bind(this));
         server.on(IPC_EVENTS.CUCUMBER, this.cypressCucumberStepListener.bind(this));
         server.on(IPC_EVENTS.PLATFORM_DETAILS, this.cypressPlatformDetailsListener.bind(this));
+        this.ipcServer = server;
```
Comment (amaanbs) [inline bin/testObservability/reporter/index.js:203]: Are we using this?

File: bin/testObservability/helper/helper.js [hunk 9]
```diff
@@ -142,6 +146,8 @@ const nodeRequest = (type, url, data, config) => {
   });
 }
 
+exports.nodeRequest = nodeRequest;
```
Comment (amaanbs) [inline bin/testObservability/helper/helper.js:149]: Needed?

File: bin/testObservability/helper/helper.js [hunk 10]
```diff
@@ -127,7 +131,7 @@ const nodeRequest = (type, url, data, config) => {
       } else if(response.statusCode != 200) {
         reject(response && response.body ? response.body : `Received response from BrowserStack Server with status : ${response.statusCode}`);
       } else {
-        try {
+        try {
```
Comment (amaanbs) [inline bin/testObservability/helper/helper.js:134]: Needed?

File: bin/testObservability/helper/helper.js [hunk 11]
```diff
@@ -17,6 +17,8 @@ const logger = require("../../helpers/logger").winstonLogger;
 const utils = require('../../helpers/utils');
 const helper = require('../../helpers/helper');
 
+const util = require('util');
```
Comment (amaanbs) [inline bin/testObservability/helper/helper.js:20]: Unused import

**Review by amaanbs — CHANGES_REQUESTED** [review]

File: bin/testObservability/helper/requestQueueHandler.js [hunk 12]
```diff
@@ -48,6 +52,15 @@ class RequestQueueHandler {
     }
   }
 
+  shutdownSync = () => {
+    this.removeEventBatchPolling('REMOVING');
+   
+    fs.writeFileSync(path.join(__dirname, PENDING_QUEUES_FILE), JSON.stringify(this.queue));
+    this.queue = [];
+    cp.spawnSync('node', [path.join(__dirname, 'cleanupQueueSync.js'), path.join(__dirname, PENDING_QUEUS_FILE)], {stdio: 'inherit'});
+    fs.unlinkSync(path.join(__dirname, PENDING_QUEUES_FILE));
```
Comment (rev-doshi) [inline bin/testObservability/helper/requestQueueHandler.js:61]: spelling fix PENDING_QUEUES_FILE

**Review by amaanbs — APPROVED** [review]

**Review by Baaryan — APPROVED** [review]

**Pull request merged**